ext/AgentsVisualizations/src/spaces/continuous.jl

```diff
@@ -0,0 +1,45 @@
+## Required
+
+Agents.agents_space_dimensionality(::ContinuousSpace{D}) where {D} = D
+
+function Agents.get_axis_limits!(model::ABM{<:ContinuousSpace})
+    e = abmspace(model).extent
+    o = zero.(e)
+    return o, e
+end
+
+## Preplots
+
+"""
+Plot heatmap according to given `heatarray`.
+Special method for models with `ContinuousSpace`.
+"""
+function heatmap!(ax, p::_ABMPlot{<:Tuple{<:ABMObservable{<:Observable{<:ABM{<:S}}}}}
```

Comment:
Actually I'm not sure how this is done due to the nesting, but maybe it is possible

Reply:
I think it should be `alias = _ABMPlot{<:Tuple{<:ABMObservable{<:Observable{<:ABM{<:S}}}} where {S}`

Reply:
Yeah for readability this would be great. I'm not sure if we can easily use `where` inside an alias and then properly access `S` from inside the functions like we normally do. Need to test that out.

Reply:
I'm also asking myself if you need to qualify it this much: why not just using `_ABMPlot`, it doesn't seem the code uses `S` most of the times, and maybe when you use it, you can just create a function which extract the spacetype from the `_ABMPlot` type e.g. 

`abmspace(::_ABMPlot{<:Tuple{<:ABMObservable{<:Observable{<:ABM{<:S}}}}) where {S} = S`

Reply:
If I understood your last suggestion correctly, then this would only save us the `where {S}` part in the function arguments. In this case here, we would still need the whole `_ABMPlot{<:Tuple{<:ABMObservable{<:Observable{<:ABM{<:ContinuousSpace}}}}}` to properly dispatch to the correct method of the `heatmap!` function. I would argue that the extra step of introducing a `abmspacetype` function isn't really all that helpful. All these functions are internal and since all of us know about the functioning of `where`, it's just easier like it currently is. But that's only my opinion. If you disagree, I can definitely make the suggested change.

Reply:
if that's the case I agree with you, but I would nonetheless try the aliasing

Reply:
The aliasing works even with something like `Alias{S} = T1{T2{T3{<:S}}}` so it's all good. I've added an alias called `ABMP{S}` that I've now used wherever appropriate. Hope this makes things a bit more legible.